managemon.c

```diff
@@ -871,7 +869,10 @@ void read_sock(struct supertype *container)
 
 	fl = fcntl(fd, F_GETFL, 0);
 	fl |= O_NONBLOCK;
-	fcntl(fd, F_SETFL, fl);
+	if (fcntl(fd, F_SETFL, fl) < 0) {
```

Comment:
`fnctl` is called two times but we are checking the result only on last attempt. I checked the man and it it not quarented that `F_GETFL` is always successful. We should check this status too.